### PR TITLE
New reference design – content refinement

### PR DESCRIPTION
**Pull request by JR-G**
## Context

We're redesigning how the references section functions. This has given us an opportunity to review the content.

This PR covers all these content changes.

## Changes proposed in this pull request
Various copy changes throughout the reference section & flow.

Section title and guidance:
|Before|After|
|---|---|
|image|image|

Section guidance:
|Before|After|
|---|---|
|image|image|

Reference review section (header & status):

|Before|After|
|---|---|
|image|image|

|Before|After|
|---|---|
|image|image|

Sending a reference guidance:

|Before|After|
|---|---|
|image|image|

Referee type selection:

|Before|After|
|---|---|
|image|image|

Entering their email:

|Before|After|
|---|---|
|image|image|

Confirming relationship:

|Before|After|
|---|---|
|image|image|

## Guidance to review

Use the review app, with the `references_selection` feature flag on and go through the start-to-finish flow of completing the reference section.

Quite a few files have been changed, but the majority of this is just small copy tweaks.

Two of the commits ([Conditionally change the review section link](https://github.com/DFE-Digital/apply-for-teacher-training/pull/4894/commits/eefff133d1fb9eefae4f54fa2bb8b6a066c61412) & [Conditionally render copy about changing selection](https://github.com/DFE-Digital/apply-for-teacher-training/pull/4894/commits/083f3e0712a7e4b6125ec0bbe344448776d1fdc8)) do introduce some logic though, so worth double checking these.

## Link to Trello card

https://trello.com/c/hZVGCcxd

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by EmmaFrith** [comment]
changes look great to me, thanks Binisa + James

@JR-G could we just make school direct (salaried) lower case in line with the style on Get Into Teaching: https://getintoteaching.education.gov.uk/ways-to-train 
?

**Comment by JR-G** [comment]
Does this need a margin above @paulrobertlloyd ?
image

File: app/views/candidate_interface/unsubmitted_application_form/show.html.erb [hunk 1]
```diff
@@ -64,6 +64,9 @@
               completed: @application_form_presenter.references_completed?,
               path: @application_form_presenter.references_selection_path,
             )) %>
+            <% if @application_form.selected_enough_references? && @application_form.application_references(&:feedback_provided?).count >= 3 %>
+              <p class="govuk-body"> You’ve selected 2 references to submit with your application already, but you can change your selection if you want.</p>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/unsubmitted_application_form/show.html.erb:68]: Let’s add a 10px margin below this paragraph:

```suggestion
              <p class="govuk-body govuk-!-margin-bottom-2">You’ve selected 2 references to submit with your application already, but you can change your selection if you want.</p>
```

**Comment by paulrobertlloyd** [comment]
@JR-G Small bug:

* View selected references (**Which 2 references do you want to include in your application?**)
* Untick a reference (i.e. make an invalid selection)
* Your previous selection isn’t shown; all checkboxes are empty.

When validating this form, we shouldn’t clear the previous answer.

File: app/components/task_list_item_references_component.html.erb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 <div class="app-task-list__content">
   <% if FeatureFlag.active?(:reference_selection) %>
     <p class="govuk-body">
-      <%= govuk_link_to t('page_titles.references_review'), references_path, 'aria-describedby': "#{t('page_titles.references_review').parameterize}-badge-id" %>
+      <% if references.count > 1 %>
```
Comment (malcolmbaig) [inline app/components/task_list_item_references_component.html.erb:4]: Does reference state matter here?

If so, perhaps this should be:
`references.first.application_form.minimum_references_available_for_selection?`

If not, we could just go with the following, and the commit message reworded accordingly:
`if references.present?`

Comment (malcolmbaig) [reply app/components/task_list_item_references_component.html.erb:4]: Whatever we decide here in terms of what link should show when, it looks like the entire if statement [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/4894/commits/f452692b25845cf7c82dc67f0f0866f72af41d6e#diff-30c2dce8fba5f45309b4ca92ab0c7886e6a7764a0a91e314dcd3dee11d84998aR57) can be absorbed by this component.

File: app/views/candidate_interface/unsubmitted_application_form/show.html.erb [hunk 3]
```diff
@@ -64,6 +64,9 @@
               completed: @application_form_presenter.references_completed?,
               path: @application_form_presenter.references_selection_path,
             )) %>
+            <% if @application_form.selected_enough_references? && @application_form.application_references.select(&:feedback_provided?).count >= 3 %>
```
Comment (malcolmbaig) [inline app/views/candidate_interface/unsubmitted_application_form/show.html.erb:67]:
```suggestion
            <% if @application_form.selected_enough_references? && @application_form.surplus_references_available_for_selection %>
```

**Review by malcolmbaig — APPROVED** [review]
Couple minor comments left inline, but otherwise LGTM 👍🏼

File: app/components/task_list_item_references_component.html.erb (hunk 2 above)
Comment (JR-G) [reply app/components/task_list_item_references_component.html.erb:4]: Note: Have added this comment to the trello card to-do list and will pick it up once we've finished the initial round of user testing

**Comment by paulrobertlloyd** [comment]
@JR-G Can we change the guidance above the ‘Add another referee’ button (on the reference review page) once 2 references have been given to:

> You can add as many referees as you like but you can only submit 2 with your application.

@EmmaFrith @B-Bonner

File: app/components/task_list_item_references_component.html.erb (hunk 2 above)
Comment (malcolmbaig) [reply app/components/task_list_item_references_component.html.erb:4]: Note for posterity: the way this link behaves will likely change following user testing. Not carding anything up at this stage, will revisit if there's anything that needs updating.

**Pull request merged**